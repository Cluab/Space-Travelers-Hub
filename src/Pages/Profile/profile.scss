.profile {
  display: flex;
  justify-content: space-between;
  font-family: Roboto, sans-serif;
  padding: 2rem;
}

.profile-rockets,
.profile-missions {
  width: 48%;
  display: flex;
  flex-direction: column;

  &>h2 {
    margin-bottom: 1rem;
    font-weight: 400;
  }

  &>.reserved-rockets,
  .missions-joined {
    border: 1px solid rgb(192 192 192);
    border-radius: 4px;

    &>p {
      padding: 2rem;
      width: 100%;
      border-bottom: 1px solid rgb(192 192 192);

      &:last-child {
        border-bottom: none;
      }
    }
  }
}