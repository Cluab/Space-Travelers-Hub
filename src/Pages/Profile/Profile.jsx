import './profile.scss';
import { useSelector } from 'react-redux';

const Profile = () => {
  const rockets = useSelector((state) => state.rockets).filter((rocket) => rocket.reserved);
  const missions = useSelector((state) => state.missions).filter((missions) => missions.member);
  return (
    <div className="profile">
      <div className="profile-missions">
        <h2 className="profileMissionH2">My Missions</h2>
        {
          missions.length > 0 && (
            <div className="missions-joined">
              {
              missions.map((mission) => <p key={mission.id} className="missionName">{mission.name}</p>)
            }
            </div>
          )
        }

      </div>
      <div className="profile-rockets">
        <h2 className="rockets-title">My Rockets</h2>
        {
          rockets.length > 0 && (
            <div className="reserved-rockets">
              {
                rockets.map(({ id, rocketName }) => <p className="rocket-name" key={id}>{rocketName}</p>)
              }
            </div>
          )
        }
      </div>
    </div>
  );
};

export default Profile;
