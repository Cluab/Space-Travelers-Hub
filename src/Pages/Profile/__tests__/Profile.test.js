import { render, act } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { configureStore } from '@reduxjs/toolkit';
import '@testing-library/jest-dom/extend-expect';
import MissionsReducer, { getMissions } from '../../../Redux/Missions/mission';
import Profile from '../Profile';
import rocketsReducer from '../../../Redux/Rockets/rockets';

const store = configureStore({ reducer: { missions: MissionsReducer, rockets: rocketsReducer } });

describe('missions component test cases', () => {
  test('should match with snapshot', async () => {
    await act(() => store.dispatch(getMissions()));
    const tree = renderer
      .create(
        <Provider store={store}>
          <Profile />
        </Provider>,
      ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
