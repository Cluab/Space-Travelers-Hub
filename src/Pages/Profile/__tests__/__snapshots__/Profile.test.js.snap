// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`missions component test cases should match with snapshot 1`] = `
<div
  className="profile"
>
  <div
    className="profile-missions"
  >
    <h2
      className="profileMissionH2"
    >
      My Missions
    </h2>
  </div>
  <div
    className="profile-rockets"
  >
    <h2
      className="rockets-title"
    >
      My Rockets
    </h2>
  </div>
</div>
`;
